app/controllers/middleware_server_controller.rb

```diff
@@ -158,6 +162,21 @@ def add_datasource
     end
   end
 
+  def jdr_download
+    mw_server = MiddlewareServer.find(from_cid(params[:id]))
+    jdr_report = mw_server.middleware_jdr_reports.find(from_cid(params[:key]))
+
+    response.headers['Content-Type'] = 'application/zip'
```

Comment:
Not that I'm aware of. There's the `mime-types` gem, but it does parsing, not output.